src/base-lazy-component.js

```diff
@@ -15,7 +15,7 @@ export default class BaseLazyComponent extends React.Component {
     return assign({}, this.props, this.resolvedData);
   }
 
-  componentWillMount() {
+  UNSAFE_componentWillMount() {
```

Comment:
I think you should simply move the code to the constructor. Otherwise, this code will need to change again when upgrading to react@17.
(In `EventListener.js` as well)

Reply:
Good point but I'm a bit scared about everything else I did here, so I'm trying to keep it as small as possible right now in case stuff breaks...

Reply:
Generally, I don't think this will have implications since this component has no state, the state is managed via ModuleRegistry.
But if it makes you feel safe, go for it ;)